controllers/cluster_controller.go

```diff
@@ -314,6 +315,39 @@ func (c *clusterDescendants) length() int {
 		len(c.workerMachines.Items)
 }
 
+func (c *clusterDescendants) descendantNames() string {
```

Comment:
I tried a number of other solutions, but this ended up being the one I liked the most. It's straightforward and did exactly what I wanted it to do. Open to suggestions here if someone feels like tinkering with this sort of refactoring.